### ISSUE TITLE
Would a simple window abstraction be a worthwhile addition?

### ISSUE THREAD

**Issue by pedrocr** [PistonDevelopers/conrod#1050]
After going through adapting the threaded winit+glium example for my application I concluded that most of that work could be abstracted away. So I wrote a window abstraction that's not specific to my app:

https://github.com/pedrocr/chimper/blob/a088385116cff5eb857cea0669b30d6752edfaa8/src/window.rs

and that makes a simple hello world app much easier to write:

https://github.com/pedrocr/chimper/blob/a088385116cff5eb857cea0669b30d6752edfaa8/src/bin/test.rs

The abstraction could be made a little cleaner still but it's already much better than writing all that stuff in your app. Would something like this be acceptable as a PR? Maybe called SimpleWindow or something?

**Comment by mitchmindtree** [comment]
Hey @pedrocr thanks for the suggestion!

We tried integrating a simple window abstraction in the past, however things quicky got out of hand in terms of maintenance and feature requests. The issues we ran into were:

1. Each backend combination has different requirements, so window and eventloop abstractions would have to be added to the backend specific code. If we were to begin supporting one of the backends (e.g. winit+glium) users of winit+gfx or piston would likely wish for the same thing for each of the different backend combinations.
2. There are many levels of support which a window abstraction may or may not add, including multiple windows, different loop rates, different ways of catching up to loop rates, wait-only event loops, custom event insertion, the kinds of events that should be available, etc.
3. Wrapper-like abstractions require being frequently updated to maintain feature-parity with the upstream crates that they wrap, which can get quite tedious.

I certainly think these issues could be solved! It's just that we have to draw a line for conrod somewhere, and I'm already struggling to keep up with conrod as it is!

I agree the current level of boilerplate in the examples can be frustrating, but I feel that the few dozen extra lines are not worth the maintenance burden and feature requests that would be involved with providing a full window abstraction from conrod itself too, especially 

Perhaps effort could be pooled into these abstractions in a downstream crate?
